frontend/packages/kubevirt-plugin/src/components/connected-vm-console/connected-vm-console.tsx

```diff
@@ -0,0 +1,138 @@
+import * as React from 'react';
+import { K8sResourceKind, PodKind } from '@console/internal/module/k8s';
+import { PodModel, ServiceModel } from '@console/internal/models';
+import { getVMStatus } from '../../statuses/vm/vm-status';
+import { VMImportKind } from '../../types/vm-import/ovirt/vm-import';
+import { V1alpha1DataVolume } from '../../types/vm/disk/V1alpha1DataVolume';
+import {
+  DataVolumeModel,
+  VirtualMachineImportModel,
+  VirtualMachineInstanceMigrationModel,
+  VirtualMachineInstanceModel,
+  VirtualMachineModel,
+} from '../../models';
+import { VMIKind, VMKind } from '../../types/vm';
+import { Firehose, FirehoseResult } from '@console/internal/components/utils';
+import { VmConsolesWrapper } from '../vms/vm-console';
+import { getLoadedData } from '../../utils';
+import { ConsoleEmptyState } from './vm-console-empty-state';
+
+const CloudVMConsole: React.FC<CloudVMConsoleProps> = ({
+  consoleType,
+  vm,
+  vmis,
+  pods,
+  migrations,
+  dataVolumes,
+  vmImports,
+}) => {
+  const loadedVM = getLoadedData(vm);
+  const loadedVMIs = getLoadedData(vmis);
+  const loadedPods = getLoadedData(pods);
+  const loadedMigrations = getLoadedData(migrations);
+  const loadedDataVolumes = getLoadedData(dataVolumes);
+  const loadedImports = getLoadedData(vmImports);
+  const vmi = loadedVMIs?.[0];
+
+  const vmStatusBundle = getVMStatus({
+    vm: loadedVM,
+    vmi,
+    pods: loadedPods,
+    migrations: loadedMigrations,
+    dataVolumes: loadedDataVolumes,
+    vmImports: loadedImports,
+  });
+
+  return (
+    <VmConsolesWrapper
+      vm={loadedVM}
+      vmi={vmi}
+      vmStatusBundle={vmStatusBundle}
+      pods={loadedPods}
+      consoleType={consoleType}
+      showActions={false}
+    />
+  );
+};
+
+type CloudVMConsoleProps = {
+  consoleType: string;
+  vm?: FirehoseResult<VMKind>;
+  vmis?: FirehoseResult<VMIKind[]>;
+  pods?: FirehoseResult<PodKind[]>;
+  migrations?: FirehoseResult<K8sResourceKind[]>;
+  dataVolumes?: FirehoseResult<V1alpha1DataVolume[]>;
+  vmImports?: FirehoseResult<VMImportKind[]>;
+  services?: FirehoseResult;
+};
+
+const ConnectedVMConsole: React.FC<ConnectedVMConsoleProps> = ({
+  consoleType,
+  namespace,
+  name,
+  isKubevirt,
+}) => {
+  const resources = [
+    {
+      kind: VirtualMachineModel.kind,
+      name,
+      namespace,
+      prop: 'vm',
+    },
+    {
+      kind: VirtualMachineInstanceModel.kind,
+      namespace,
+      isList: true,
+      prop: 'vmis',
+      optional: true,
+      fieldSelector: `metadata.name=${name}`,
+    },
+    {
+      kind: PodModel.kind,
+      namespace,
+      isList: true,
+      prop: 'pods',
+    },
+    {
+      kind: VirtualMachineInstanceMigrationModel.kind,
+      isList: true,
+      namespace,
+      prop: 'migrations',
+    },
+    {
+      kind: VirtualMachineImportModel.kind,
+      isList: true,
+      namespace,
+      prop: 'vmImports',
+      optional: true,
+    },
+    {
+      kind: DataVolumeModel.kind,
```

Comment:
these tagging resources are becoming quite a big pain. We should refactor how we use them.